pykafka/producer.py

```diff
@@ -111,15 +158,112 @@ def __repr__(self):
             id_=hex(id(self))
         )
 
-    def _send_request(self, broker, req, attempt):
+    def __enter__(self):
+        """Context manager entry point - start the producer"""
+        self.start()
+        return self
```

Comment:
You're calling `self.start()` twice. In `__init__` and here. I'd leave it in `__init__` and remove this one.